h2o-algos/src/main/java/hex/tree/SharedTreeMojoWriter.java

```diff
@@ -38,6 +38,34 @@ protected void writeModelData() throws IOException {
       writekv("calib_method", "platt");
       writekv("calib_glm_beta", beta);
     }
+    writekv("_genmodel_encoding", model.getGenModelEncoding());
+    String[] origNames = model._output._origNames;
+    if (origNames != null) {
+      int nOrigNames = origNames.length;
+      writekv("_n_orig_names", nOrigNames);
+      startWritingTextFile("_orig_names");
+      for (String name : origNames) {
+        writeln(name, true);
+      }
+      finishWritingTextFile();
+    }
+    if (model._output._origDomains != null) {
+      int nOrigDomainValues = model._output._origDomains.length;
+      writekv("_n_orig_domain_values", nOrigDomainValues);
+      for (int i=0; i < nOrigDomainValues; i++) {
+        String[] currOrigDomain = model._output._origDomains[i];
+        writekv("_m_orig_domain_values_" + i, currOrigDomain == null ? 0 : currOrigDomain.length);
+        if (currOrigDomain != null) {
```

Comment:
it would be nice to refactor this start - for() - stop into a method, maybe writeDomains() can use it too

Reply:
fixed